### PR TITLE
Text: Add new semantic colors

### PR DESCRIPTION
**Pull request by ayeshakmaz**
#### What changed?

Add new values for Text that follow semantic design tokens

#### Why?

Instead of allowing all colors on Text, we're updating the color prop to use semantic color names that follow our new design tokens. The old values will be deprecated in a future PR.

### Links

- [Jira](https://jira.pinadmin.com/browse/GESTALT-3944)

### Checklist

- [ ] Added unit and Flow Tests
- [ ] Added documentation + accessibility tests
- [ ] Verified accessibility: keyboard & screen reader interaction
- [ ] Checked dark mode, responsiveness, and right-to-left support
- [ ] Checked stakeholder feedback (e.g. Gestalt designers)

### CODE REVIEW

**Comment by christianvuerings** [comment]
@ayeshakmaz nit - what's the migration timeline for these? E.g. right now it's confusing as to when to use `red` or `error`.

image

**Comment by ayeshakmaz** [comment]
@christianvuerings we're recommending no one us the old colors going forward, so people should be using error going forward. Next sprint I'll create a codemod and run it in Pinboard to convert current mappings over, but we need to reconcile the blue color first

**Review by AlbertCarreras — APPROVED** [review]
LGTM

**Comment by AlbertCarreras** [comment]
LGTM but Flow is complaining...
Screen Shot 2022-03-08 at 9 38 01 PM

**Review by AlbertCarreras — CHANGES_REQUESTED** [review]
The Flow issue

**Comment by ayeshakmaz** [comment]
@AlbertCarreras let me know if this looks good to go now, thanks!

File: docs/pages/color_usage.js [hunk 1]
```diff
@@ -43,7 +43,7 @@ export default function ColorUsagePage(): Node {
       >
         <ColorTile
           description="Primary base"
-          textColor="white"
+          textColor="light"
```
Comment (ayeshakmaz) [inline docs/pages/color_usage.js:46]: Changes in this file make use of the new values as a sort of test

File: docs/pages/text.js [hunk 2]
```diff
@@ -5,34 +5,39 @@ import PageHeader from '../components/PageHeader.js';
 import Page from '../components/Page.js';
 import GeneratedPropTable from '../components/GeneratedPropTable.js';
 import docgen, { type DocGen } from '../components/docgen.js';
+import MainSection from '../components/MainSection.js';
```
Comment (AlbertCarreras) [inline docs/pages/text.js:8]: cleanup changes, unrelated. (for review tracking)

File: docs/pages/text.js [hunk 3]
```diff
@@ -41,63 +46,85 @@ export default function TextPage({ generatedDocGen }: {| generatedDocGen: DocGen
...
+    <Text color="warning" size="400">Warning</Text>
+    <Text color="error" size="400">Error</Text>
+    <Text color="shopping" size="400">Shopping</Text>
+    <Box color="primary" padding={1}>
+      <Text color="light" size="400">Light</Text>
+    </Box>
+    <Box color="infoWeak" padding={1}>
+      <Text color="dark" size="400">Dark</Text>
+    </Box>
+  </Flex>
```
Comment (ayeshakmaz) [inline docs/pages/text.js:78]: This is the main update on this page

File: packages/gestalt/src/Text.js [hunk 4]
```diff
@@ -106,10 +115,22 @@ export default function Text({
   underline = false,
   weight = 'normal',
 }: Props): Node {
+  const semanticColors = [
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Text.js:118]: suggestion

i think we could prevent mismatches if we move `semanticColors` to textTypes.js and then spread it into `allowedColors`

File: packages/gestalt/src/Typography.css [hunk 5]
```diff
@@ -31,6 +31,44 @@
   font-size: var(--font-size-600);
 }
 
+/* colors */
```
Comment (AlbertCarreras) [inline packages/gestalt/src/Typography.css:34]: nit, worth adding _semantic_ here? just a thought

File: packages/gestalt/src/textTypes.js [hunk 6]
```diff
@@ -23,6 +23,15 @@ export const allowedColors = [
...
+  'success',
+  'error',
+  'warning',
+  'inverse',
+  'shopping',
+  'light',
+  'dark',
 ];
 
 export type Color =
```
Comment (AlbertCarreras) [inline packages/gestalt/src/textTypes.js:37]: are we using this export type Color anywhere? maybe we could remove it. With autogenerated prop tables, we are inlining. could be worth checking

**Review by AlbertCarreras — APPROVED** [review]
LGTM

File: docs/pages/text.js [hunk 7]
```diff
@@ -41,63 +46,85 @@ export default function TextPage({ generatedDocGen }: {| generatedDocGen: DocGen
...
+    </Text>
+  </Flex>
+</Box>
+        `}
+          />
+        </MainSection.Subsection>
+        <MainSection.Subsection
+          title="Sizes"
+          description={`
+          You can apply \`size\` options to define the size of the text. These font sizes follow those available through our [Design Tokens](/design_tokens#Font-size). If your text needs to be a [semantic heading (H1-H6)](/https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements), use [Heading](/heading) instead.
```
Comment (AlbertCarreras) [inline docs/pages/text.js:122]: ALERT link  is broken

Comment (AlbertCarreras) [reply docs/pages/text.js:122]: [semantic heading (H1-H6)](**/**https://developer.mozilla.org/en-US/docs/Web/HTML/Element/Heading_Elements)

File: packages/gestalt/src/textTypes.js (hunk 6 above)
Comment (ayeshakmaz) [reply packages/gestalt/src/textTypes.js:42]: I'll look into this when we do this for Heading

**Pull request merged**